### PR TITLE
feat: mask on antennas

### PR DESCRIPTION
**Pull request by steven-murray**
Adds the ability to set a mask on which antennas to use (simple min/max distance from origin).

This is useful for arrays with a "core" and "outriggers" if you just want to use one part of the array, for instance.

### CODE REVIEW

File: py21cmsense/observatory.py [hunk 1]
```diff
@@ -130,6 +133,20 @@ def from_yaml(cls, yaml_file):
...
+        antpos = antpos[np.sum(np.square(antpos), axis=1) < max_antpos ** 2]
+
+        if max_antpos < np.inf:
+            logger.info(
+                f"Removed {_n - len(antpos)} antennas using given max_antpos={max_antpos} m."
+            )
+
+        # If we get only East and North coords, add zeros for the UP direction.
+        if antpos.shape[1] == 2:
+            antpos = np.hstack((antpos, np.zeros((len(antpos), 1))))
```
Comment (mkolopanis) [inline py21cmsense/observatory.py:148]: While you're adding a maximum antenna position, is there a desire for a minimum one too? or a minimum uvw range maybe?

**Review by mkolopanis — COMMENTED** [review]
Nice and simple, one very quick question below.

File: py21cmsense/observatory.py (hunk 1 above)
Comment (steven-murray) [reply py21cmsense/observatory.py:148]: There's already a `bl_min` and `bl_max` for the `Observation` (in meters units). It makes sense to have a minimum antenna position. I'll add that.

Comment (steven-murray) [reply py21cmsense/observatory.py]: OK, did that. But also realized it was only using those parameters from a YAML, not when instantiating the object directly. So I reshuffled it to use those parameters at the top-level.

Then I did `pre-commit autoupdate` and it changed a whole bunch of files. Sorry.

File: tests/test_observatory.py [hunk 2]
```diff
@@ -84,3 +85,21 @@ def test_grid_baselines(bm):
...
+    a = Observatory(
+        antpos=np.array([np.linsapce(0, 50, 11), np.zeros(11), np.zeros(11)]),
+        beam=bm,
+        min_antpos=7,
+    )
+
+    assert len(a.antpos) == 9
+
+    a = Observatory(
+        antpos=np.array([np.linsapce(0, 50, 11), np.zeros(11), np.zeros(11)]),
```
Comment (mkolopanis) [inline tests/test_observatory.py:100]: some easy typos here causing test failures

Comment (steven-murray) [reply tests/test_observatory.py]: oh good. OK that's fixed.

**Review by mkolopanis — APPROVED** [review]
Thanks for the changes!

**Pull request merged**